readthedocs/doc_builder/python_environments.py

```diff
@@ -615,7 +615,7 @@ def _append_core_requirements(self):
 
             for item in dependencies:
                 if isinstance(item, dict) and 'pip' in item:
-                    pip_requirements.extend(item.get('pip', []))
+                    pip_requirements.extend(item.get('pip') or [])
```

Comment:
Worth a comment here.

Reply:
@stsewd the comment you added in the description of the PR is a good one. "pip can be `None`" says less than the description, where you explained _the reason_ of `pip` being `None` which adds good context to the problem.